HACKING.md

```diff
@@ -0,0 +1,88 @@
+# Hacking
+
+## `make-prs.py`
+
+The `make-prs.py` script (only exists in the main branch) does the following:
+
+- Get the diff from a PR;
+- find out which chapter that PR is about;
+- create branches (in your forked repo), merge the diff into all following chapters, commit, and push;
+- create PRs to the upstream.
+
+### Configuration
+
+- GITHUB_TOKEN env var is set.
+
+### Dependencies
+
+- `git` CLI
+- `tox`
+- `PyGithub`, `requests` (both are already automatically managed by `tox`)
+
+### Usage
+
+```console
+usage: make-prs.py [-h] [-u UPSTREAM_OWNER] -f FORK_OWNER [-ur UPSTREAM_REPO_NAME]
+                   [-fr FORK_REPO_NAME] -p PULL_REQUEST_NUMBER [-i]
+
+options:
+  -h, --help            show this help message and exit
+  -u UPSTREAM_OWNER, --upstream-owner UPSTREAM_OWNER
+                        The upstream owner of the repository, defaults to `canonical`.
+  -f FORK_OWNER, --fork-owner FORK_OWNER
+                        The forked repo's owner, mandatory. Normally, it should be your own
```

Comment:
Given that this is mandatory, shouldn't it be an argument rather than an option? Or could we default it to `getpass.getuser()`? That won't always be correct (it would not be for me by default!) but will be in some cases, and it's then also easily wrapped in a tool by setting one of the environment variables.

Reply:
`getpass.getuser()` won't do it for me either.

And since a PR number is also required, having two positional arguments like `python3 make-prs.py IronCore864 1 -xxx` seems less readable than plainly all options (some mandatory) like `python3 make-prs.py --fork-owner IronCore864 --pull-request-number 1 --ignore-conflicts`.

Or, maybe it doesn't matter that much because it's most likely executed by `tox -e make-prs` anyway.

So I will keep it as it is, I hope this is fine. I am closing the other comment related to this as well.